spec/abci/abci++_comet_expected_behavior.md

```diff
@@ -224,3 +227,35 @@ As for the new methods:
 Finally, `Commit`, which is kept in ABCI++, no longer returns the `AppHash`. It is now up to
 `FinalizeBlock` to do so. Thus, a slight refactoring of the old `Commit` implementation will be
 needed to move the return of `AppHash` to `FinalizeBlock`.
+
+## Accomodating for vote extensions
+
+In a manner transparent to the application, CometBFT ensures the node is provided with all
+the data it needs to participate in consensus. 
+
+In the case of recovering from a crash, or joining the network via state sync, CometBFT will make
+sure the node acquires the neccessary vote extensions before switching to conensusus. 
+
+If a node is already in consensus but falls behind, during catch-up, CometBFT will provide the node with 
+vote extensions from past heights by storing the extensions within `ExtendedCommit` for old heights.
+
+We realize this is sub-optimal due to the increase in storage needed to store the extensions, we are 
+working on an optimization of this implementation which should eliviate this concern. The decision to store 
+historical commits and potential optimizations, are discussed in detail in [RFC-100](./../../docs/rfc/rfc-100-abci-vote-extension-propag.md#current-limitations-and-possible-implementations)
+
+## Handling upgrades to ABCI 2.0 
+
+If applications upgrade to ABCI 2.0, CometBFT internally ensures that the [application setup](./abci%2B%2B_app_requirements.md#application-configuration-required-to-switch-to-abci-20) is reflected in its opertaion. 
+
+Namely, upon saving the block for a given height *h* in the block store at decision time
+- if *h ≥ h<sub>e</sub>*, the corresponding extended commit is saved as  well
```

Comment:
You probably need to redefine *h<sub>e</sub>* here (since you can't assume the reader is reading the whole spec)